website/content/en/highlights/2022-03-22-0-21-0-upgrade-guide.md

```diff
@@ -50,6 +51,67 @@ Updating your subscriptions is as simple as renaming `patterns` to
 +  outputEventsByComponentIdPatterns(outputsPatterns: [...])
 ```
 
+#### GraphQL API `EventNotification` type changes {#api-event-notification-changes}
+
+As part of adding a new notification (`InvalidMatch`) to warn users against
+attempting invalid matches, we've reshaped the `EventNotification` type for
+easier querying and future extensibility.
+
+Previously, the `EventNotification` type consisted simply of a `pattern` and
+plain enum describing the notification.
+
+```graphql
+type EventNotification {
+  pattern: String!
+  notification: EventNotificationType!
+}
+```
+
+While this worked well for simple notifications like `Matched` and `NotMatched`,
+it was awkward to extend to new notifications, like `InvalidMatch`, which may
+want to include more information than `pattern`. Thus, the new
+`EventNotification` type takes the following form:
+
+```graphql
+type EventNotification {
```

Comment:
Nit: It might be nice to include the definition or `Notification` here too, to give a more complete picture of the API changes.